### ISSUE TITLE
inspec with chef vault

### ISSUE THREAD

**Issue by AnotherNerdHere** [inspec/inspec#609]
We use chef vault in most of of our cookbooks for sensitive user ids and passwords and other info.
I need a way to be able to use the valult in the inspec profiles but I don't see it documented anywhere. passwords in clear text is unacceptable and will not be allowed by the powers that be.

**Comment by chris-rock** [comment]
@AnotherNerdHere Thanks for the feedback. And YES, we agree that sensitive data should not be part of profiles. That is one of the reasons, why we have not yet reached 1.0. For the next iterations we envision an attributes interface that allows you to call inspec with the required parameters. Once that is done, an integration with chef-vault or other secrets systems is possible. For now, you could do something like:

``` ruby
require 'json'
secrets_file = File.read('/secrets/secrets.json')
secrets_hash = JSON.parse(secrets_file)

sql = mysql_session(secrets_hash['mysql_user'],secrets_hash['mysql_password'])
describe sql.query('show databases like \'test\';') do
  its(:stdout) { should_not match(/test/) }
end

```

Just ensure the secrets file is available where you execute the profile (workstation, not scanned node). Does that  help?

**Comment by AnotherNerdHere** [comment]
@chris-rock Thanks for the reply and code. I'll give that a try. the biggest hiccup may be the last sentence, the location of the secrets file.

**Comment by chris-rock** [comment]
@AnotherNerdHere I agree this is not a perfect solution yet. The improvement of secret management is high on our list.

**Comment by johnypony3** [comment]
this issue is now three years old. can you update this please?

**Comment by AnotherNerdHere** [comment]
Apologies. I am no longer with that company and no longer using _any_ chef product. I would say close it

**Comment by clintoncwolfe** [comment]
An attribute/input system as described above was indeed implemented. Our current work in this area is in making the input system modular.  See #3666 .

**Comment by kekaichinose** [comment]
__What’s happening? Why was this issue closed?__
This issue was closed due to some much needed review of legacy issues or issues that were spawned in older versions of InSpec, i.e. < v3.

__Why do I care?__
You would care about this if this was an issue that you are still seeing and/or feel needs to be addressed in the current version of InSpec.

__What do I need to do?__
If this issue is no longer important, no further action is necessary. However, if you think this is something that should be addressed, please open a new issue and refer to the original issue in the description.
